eslint.config.js

```diff
@@ -0,0 +1,316 @@
+					]
+				}
+			]
+		}
+	},
+
+	{
+		files: [ "src/wrapper-esm.js" ],
+		languageOptions: {
+			globals: {
+				jQuery: false
+			}
+		},
+		rules: {
+			"no-unused-vars": "off",
+			indent: [
+				"error",
+				"tab",
+				{
+
+					// Unlike other codes, "wrapper.js" is implemented in UMD.
+					// So it required a specific exception for jQuery's UMD
+					// Code Style. This makes that indentation check is not
+					// performed for 1 depth of outer FunctionExpressions
+					ignoredNodes: [
+						"Program > FunctionDeclaration > *"
+					]
+				}
+			]
+		}
+	},
+
+	{
+		files: [ "src/exports/amd.js" ],
+		languageOptions: {
+			globals: {
+				define: false
+			}
+		}
+	},
+
+	// Tests
+	{
+		files: [
+			"test/**"
+		],
+		ignores: [
+			"test/data/jquery-1.9.1.js",
+			"test/data/badcall.js",
+			"test/data/badjson.js",
+			"test/data/support/csp.js",
+			"test/data/support/getComputedSupport.js",
+			"test/data/core/jquery-iterability-transpiled.js"
+		],
+		languageOptions: {
+			globals: {
+				...globals.browser,
+				require: false,
+				Promise: false,
+				Symbol: false,
+				trustedTypes: false,
+				QUnit: false,
+				ajaxTest: false,
+				testIframe: false,
+				createDashboardXML: false,
+				createWithFriesXML: false,
+				createXMLFragment: false,
+				includesModule: false,
+				moduleTeardown: false,
+				url: false,
+				q: false,
+				jQuery: true,
+				sinon: true,
+				amdDefined: true,
+				fireNative: true,
+				Globals: true,
+				hasPHP: true,
+				isLocal: true,
+				supportjQuery: true,
+				originaljQuery: true,
+				$: true,
+				original$: true,
+				baseURL: true,
+				externalHost: true
+			}
+		},
+		rules: {
+			...jqueryConfig.rules,
+			strict: [ "error", "function" ],
+
+			// See https://github.com/eslint/eslint/issues/2342
+			"no-unused-vars": "off",
+
+			// Too many errors
+			"max-len": "off",
+			camelcase: "off",
+			"one-var": "off"
+		}
+	},
+
+	{
+		files: [
+			"test/data/core/jquery-iterability-transpiled-es6.js"
+		],
+		languageOptions: {
+			sourceType: "script"
+		}
+	},
+
+	{
+		files: [
+			"test/unit/deferred.js"
+		],
+		rules: {
+
+			// Deferred tests set strict mode for certain tests
+			strict: "off"
+		}
+	},
+
+	{
+		files: [
+			"test/node_smoke_tests/commonjs/**",
+			"test/node_smoke_tests/module/**",
+			"test/promises_aplus_adapters/**",
+			"test/middleware-mockserver.cjs"
+		],
+		languageOptions: {
+			globals: {
+				...globals.node,
+				...globals.es2021
+			}
+		},
+		rules: {
+			strict: [ "error", "global" ]
+		}
+	},
+
+	{
+		files: [
+			"build/**"
+		],
+		languageOptions: {
+			globals: {
+				...globals.node,
+				...globals.es2021
+			},
+			sourceType: "commonjs"
+		},
+		rules: {
+			...jqueryConfig.rules,
+			strict: [ "error", "global" ]
+		}
+	},
+
+	{
+		files: [
+			"dist/jquery.js",
+			"dist/jquery.slim.js",
+			"dist-module/jquery.module.js",
+			"dist-module/jquery.slim.module.js"
```

Comment:
These two are repeated below. Also, I don't think you can group `dist` & dist-module` files since the latter ones are modules. The fact that it works suggests to me something's wrong about the config. Shouldn't we explicitly declare `sourceType` for all the file groups?

Reply:
I considered that, but learned that the sourceType is inferred from the package.json. Specifically, `"type": "module"` set all js files to sourceType "module". So, we only need to specify `sourceType` when we want it to differ from the default. I actually meant to remove the repeated lines below this.

Reply:
Nice!